temporalio/converter.py

```diff
@@ -732,16 +732,17 @@ def _type_hints_from_func(
     # the func. This mimics inspect logic inside Python.
     if (
         not inspect.isfunction(func)
-        and not isinstance(func, type)
         and not isinstance(func, _non_user_defined_callables)
         and not isinstance(func, types.MethodType)
     ):
-        # Callable instance
-        call_func = getattr(type(func), "__call__", None)
+        # Class type or Callable instance
+        tmp_func = func if isinstance(func, type) else type(func)
```

Comment:
Would have liked a better variable name than `tmp_func` but this is fine for now